android/src/main/java/com/reactnativecommunity/webview/URLUtil.java

```diff
@@ -84,9 +84,12 @@ public static final String guessFileName(
             }
         }
 
-        // Finally, if couldn't get filename from URI, get a generic filename
         if (filename == null) {
+            // If couldn't get filename from URI, get a generic filename
             filename = "downloadfile";
+        } else {
+            // Sanitize filename by replacing invalid characters with "_"
+            filename = filename.replaceAll("[\\\\/%\"]", "_");
```

Comment:
Would it make sense to move this logic to where the `Request` is configured in `RNCWebViewManager`?

As the comments at the top of this file indicate, it is taken from https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/webkit/URLUtil.java and mostly unmodified.

There is one exception, but that is explicitly noted in the comments and would not have worked outside of this file... whereas this sanitization change seems like it would be safe to relocate and run on the string returned by `guessFileName`?

(if not, then I'd ask that it at least be added to the comments calling out modifications to the file)

Reply:
Yep, if `URLUtil` was an original file of `react-native-webview` I would totally agree with your logic, I'm just biased towards keeping it a "pure" copy of the Android WebKit version as much as possible...